packages/grafana-ui/src/components/Forms/Form.mdx

```diff
@@ -106,11 +108,38 @@ import { Forms } from '@grafana/ui';
   </>
 )}
 </Forms.Form>
+```
+Note that when using `Forms.InputControl`, it expects the name of the prop that handles input change to be called `onChange`.
+If the property is named differently for any specific component, additional `onChangeName` prop has to be provided, specifying the name.
+Additionally, the value to for the `onChange` prop is passed as an array. Check [react-hook-form docs](https://react-hook-form.com/api/#Controller)
+for more prop options.
+
+```jsx
+{/* In case of Select the value has to be returned as an object with a `value` key for the value to be saved to form data */}
+const onSelectChange = ([value]) => {
+  // ...
+  return { value };
+}
+
+<Field label="SelectExample">
+  <Forms.InputControl
+    as={Select}
+    control={control}
+    name="select"x
+    onSelectChange={onSelectChange}
+    {/* Pass the name of the on change handler */}
+    onChangeName='onSelectChange'
+  />
+</Field>
+
 ```
 
 ### Default values
 
-Default values of the form can be passed either via `defaultValues` property on the `Form` element, or directly on form's input via `defaultValue` prop:
+Default values of the form can be passed either via `defaultValues` property on the `Form` element, or directly on
+form's input via `defaultValue` prop.
+Note that changing/updating `defaultValues` passed to the form will reset the form's state, which might be undesirable in
+case it has both controlled and uncontrolled components. In that case it's better to pass `defaultValue` to each form component separately.
```

Comment:
Great note!

Reply:
:) I'm actually not sure what's the reasoning for this behavior, since it'd seem to me that default values would be only used once on init and not trigger form reset when changed.

Reply:
Ok, so we added this behavior few weeks ago, to be able to provide changed default values. Maybe it's not the best decision. But does changing defaultValue's value on input level makes the form update it? Because by default, if you change the default values on the form level, those are not updated in the UI.

Reply:
The specific case I had was when `defaultValues` were passed directly from the Redux state. Updating that Redux state caused all the uncontrolled form values to be reset to defaults. 

This issue is only relevant when there's a combination of form values that are uncontrolled and controlled ones, the values for which come from Redux/component state